src/main/java/org/folio/processing/mapping/mapper/writer/marc/MarcBibRecordModifier.java

```diff
@@ -182,7 +208,10 @@ private boolean fieldLinked(DataField dataField) {
     }
 
     return bibAuthorityLinksKept.stream()
-      .filter(link -> link.getBibRecordTag().equals(dataField.getTag()))
+      .filter(link -> linkingRules.stream()
+        .map(LinkingRuleDto::getId)
+        .collect(Collectors.toList())
+        .contains(link.getLinkingRuleId()))
```

Comment:
This will always return all links.
If you want to get only links related to a field - you need to filter linking rules by field code first
`var rulesForTag = linkingRules.stream...`
`return bibAuthorityLinksKept.stream()
      .filter(link -> rulesForTag.stream...`